lib/xcodeproj/scheme/environment_variables.rb

```diff
@@ -0,0 +1,117 @@
+require 'xcodeproj/scheme/xml_element_wrapper'
+
+module Xcodeproj
+  class XCScheme
+    VARIABLES_NODE = 'EnvironmentVariables'
+    VARIABLE_NODE = 'EnvironmentVariable'
+
+    # This class wraps the EnvironmentVariables node of a .xcscheme XML file. This
+    # is just a container of EnvironmentVariable objects. It can either appear on a
+    # LaunchAction or TestAction scheme group.
+    #
+    class EnvironmentVariables < XMLElementWrapper
+      # @param [REXML::Element,Array{EnvironmentVariable},Array{Hash{String => String}}] node_or_target
+      #   The 'EnvironmentVariables' XML node that this object represents.
+      #   If nil, it will create a default XML node to use.
+      #
+      def initialize(node_or_target = nil)
+        create_xml_element_with_fallback(node_or_target, VARIABLES_NODE) do
+          @all_variables = []
+          node_or_target.each { |var| add_variable(var) } unless node_or_target.nil?
+        end
+      end
+
+      # @return [Array{EnvironmentVariable}]
+      #   The key value pairs currently set in @xml_element
+      #
+      def all_variables
+        @all_variables ||= @xml_element.get_elements(VARIABLE_NODE).map { |variable| EnvironmentVariable.new(variable) }
+      end
+
+      # @param [EnvironmentVariable,Hash{String => String}] variable
+      #   The key value pairs currently set in @xml_element
```

Comment:
"currently set"? `add_variable` is to add new variables not currently in the node, right?